bindings/python/test/test_pandas.py

```diff
@@ -219,3 +224,45 @@ def test_find_decimal128(self):
 
         table = find_pandas_all(self.coll, {}, schema=self.schema)
         pd.testing.assert_frame_equal(expected, table)
+
+
+class TestNulls(TestNullsBase):
+    def find_fn(self, coll, query, schema):
+        return find_pandas_all(coll, query, schema=schema)
+
+    def equal_fn(self, left, right):
+        left = left.fillna(0).replace(-0b1 << 63, 0)  # NaN is sometimes this
```

Comment:
What is `-0b1 << 63` and why is it only sometimes?

Reply:
On some instances of GitHub Actions, it would fail with the NaNs replaced with a bit string of all 1s instead of 0s and the negative sign (example: https://github.com/mongodb-labs/mongo-arrow/runs/7550724404?check_suite_focus=true).

Reply:
@blink1073 does this make sense to you?

Reply:
It sounds like an overflow error in pandas, that's a head-scratcher.

Reply:
That value is actually the minimum int64 value:
```
  File ".../bindings/python/test/test_arrow.py", line 498, in test_int_handling
    self.equal_fn(res_table["int64"], table_write["int64"])
...
AssertionError: Series are different

Series values are different (50.0 %)
[index]: [0, 1, 2, 3]
[left]:  [0, -9223372036854775808, 2, -9223372036854775808]
[right]: [0.0, 0.0, 2.0, 0.0]
```

```
>>> print(bin(-9223372036854775808))
-0b1000000000000000000000000000000000000000000000000000000000000000
```

Reply:
I'm fine if we merge with this hack but could you open a ticket so we can remember to report it as a bug (to pandas or possibly arrow)?

Reply:
Opened a ticket here: https://jira.mongodb.org/browse/ARROW-113